site/content/guides/scaling-your-monitoring-setup.md

```diff
@@ -0,0 +1,181 @@
+
+Suddenly, what started as a simple setup turns into something fragile: hard to maintain, easy to break, and impossible to audit.
+
+That’s when it’s time to shift from clicking around in the UI to managing your monitoring like the rest of your infrastructure and application: **as code**.
+
+![checkly platform screenshot](/guides/images/scaling-monitoring-setup.png)
+
+## Why Manual Monitoring Doesn’t Scale
+
+As systems grow, the cracks in a UI-based setup start to show. The workflow that once felt simple now gets in the way of moving fast and staying reliable.
+
+For example:
+
+- **Every new service adds overhead.** You’re no longer just spinning up one or two checks—you’re repeating the same setup steps dozens of times for every environment, feature, and team. The time adds up quickly, and it’s time not spent on building.
+- **Small changes become a huge chore.** Something as minor as a new URL path or auth header might require updating hundreds of checks by hand. Without version control or bulk edits, it’s easy to make mistakes—or avoid making changes altogether.
+- **You lose visibility into what’s happening.** When checks are created or edited through the UI, there’s no clear history. If something gets deleted or misconfigured, you’re left guessing who did it and why.
+- **Team collaboration becomes fragile.** As more people touch monitoring, the risk of accidental overwrites or conflicting edits increases. There’s no way to track intent, review changes, or roll things back safely.
+- **UI-based tools hit a ceiling.** Record-and-replay flows might work for simple use cases, but they often struggle with dynamic content, authentication, or multi-step flows. You end up spending more time maintaining the tool than getting value from it.
+
+### What Teams Really Want
+
+Eventually, teams start looking for something better—something that fits into the way they already build and deploy software:
+
+- A monitoring setup that’s versioned, reusable, and testable
+- A way to automate checks as part of CI/CD pipelines
+- Visibility into who changed what and when
+- A workflow that supports scale, not fights against it
+
+That’s the promise of [**Monitoring as Code](https://www.checklyhq.com/learn/monitoring/monitoring-as-code/)** and why more teams adopt it as they grow.
+
+## Monitoring as Code: A Better Way to Scale
+
+Monitoring as Code addresses all these problems. Just like Infrastructure as Code transformed the way teams manage cloud resources, Monitoring as Code gives you a way to manage your monitors with consistency, structure, and automation.
+
+Instead of relying on manual UI edits, you define your checks, alert channels, and groups as code. They live in your Git repo, are version-controlled, reviewed via pull requests, and deployed through CI/CD—just like the rest of your infrastructure.
+
+Here’s an example of a simple browser monitor running a Playwright script defined with code:
+
+```jsx
```

Comment:
Technically this is `ts`. Doesn't this work?

Reply:
However, we also have code blocks with file names. If you'd use these then you could remove the file comment as the first line and it would look nicer.